pkg/kubelet/kubelet.go

```diff
@@ -691,7 +694,12 @@ func NewMainKubelet(ctx context.Context,
 	klet.mirrorPodClient = kubepod.NewBasicMirrorClient(klet.kubeClient, string(nodeName), nodeLister)
 	klet.podManager = kubepod.NewBasicPodManager()
 
-	klet.statusManager = status.NewManager(klet.kubeClient, klet.podManager, klet, kubeDeps.PodStartupLatencyTracker)
+	broadcaster := pods.NewBroadcaster()
+	klet.podsServer = pods.NewPodsServer(broadcaster)
+	podSubscribers := []subscription.PodUpdateSubscriber{klet.podsServer}
+	statusSubscribers := []subscription.StatusUpdateSubscriber{klet.podsServer}
```

Comment:
Can we hide this under the feature gate?

Reply:
Done.